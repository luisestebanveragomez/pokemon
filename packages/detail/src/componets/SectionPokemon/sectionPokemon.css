.pokemon {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  position: relative;
  z-index: 2;
}

.pokemon__img {
  width: 250px;
  height: auto;
  margin-bottom: 15px;
}

.pokemon__category {
  text-transform: uppercase;
  margin-left: 10px;
  opacity: 0.8;
  font-weight: 500;
  color: var(--white)
}

.pokemon__name {
  font-weight: bold;
  font-size: 4rem;
  line-height: 60px;
  color: var(--white);
  text-transform: capitalize;
}

.pokemon__number {
  position: absolute;
  font-weight: 900;
  font-size: 16vmax;
  color: rgba(255, 255, 255, 0.18);
  user-select: none;
  left: 50%;
  top: 50%;
  z-index: -1;
  transform: translate(-50%, -50%);
}

@media screen and (min-width: 800px) {
  .pokemon__number {
    font-size: 20vmax;
  }
}