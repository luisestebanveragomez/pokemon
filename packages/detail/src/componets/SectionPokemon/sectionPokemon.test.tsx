import { render } from '@testing-library/react';
import { SectionPokemon } from './SectionPokemon';

describe('SectionPokemon Component', () => {
  it('Should Be Render', () => {
    const { container } = render(<SectionPokemon id={'12'} detail={null} />)
    expect(container).toMatchSnapshot()
  })

  it('Should Be Validate Number', () => {
    const { container } = render(<SectionPokemon id={'12'} detail={null} />)
    const number = document.querySelector('.pokemon__number')

    expect(number.textContent).toEqual('#12')
  })
})