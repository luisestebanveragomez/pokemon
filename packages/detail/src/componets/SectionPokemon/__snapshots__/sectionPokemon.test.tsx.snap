// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionPokemon Component Should Be Render 1`] = `
<div>
  <section
    class="relative"
  >
    <figure
      class="pokemon"
    >
      <img
        alt=""
        class="pokemon__img"
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/12.svg"
      />
      <figcaption>
        <span
          class="pokemon__category"
        />
        <h2
          class="pokemon__name"
        />
      </figcaption>
    </figure>
    <div
      class="pokemon__number"
    >
      #
      12
    </div>
  </section>
</div>
`;
